### PR TITLE
Discovery Engine cluster role

### LINKED ISSUE #642: remove cluster-admin clusterrole for discovery-engine
Currently, [cluster-admin role is used](https://github.com/kubearmor/discovery-engine/blob/b7a92bc6d110b6338ef5913c29454350459dd1b9/deployments/k8s/deployment.yaml#L153) for discovery-engine. This is a security risk.

### PR DESCRIPTION
**Pull request by vishnusomank**
- replaced cluster-admin role with identified cluster roles to achieve least privilege
Signed-off-by: Vishnu Soman <vishnu@accuknox.com>

Fixes: https://github.com/accuknox/discovery-engine/issues/642

### CODE REVIEW

File: deployments/k8s/deployment.yaml [hunk 1]
```diff
@@ -56,7 +56,7 @@ data:
         operation-mode: 1                         # 1: cronjob | 2: one-time-job
         cron-job-time-interval: "0h0m10s"         # format: XhYmZs
         operation-trigger: 5
-        network-log-from: "kubearmor"             # db|hubble|feed-consumer|kubearmor
```
Comment (Ankurk99) [inline deployments/k8s/deployment.yaml]: why did you remove kubearmor?

File: deployments/k8s/deployment.yaml [hunk 2]
```diff
@@ -65,7 +65,6 @@ data:
       system:
         operation-mode: 1                         # 1: cronjob | 2: one-time-job
         cron-job-time-interval: "0h0m10s"         # format: XhYmZs
-        operation-trigger: 5
```
Comment (Ankurk99) [inline deployments/k8s/deployment.yaml]: we need an operation-trigger, please keep this.

**Review by Ankurk99 — CHANGES_REQUESTED** [review]
please find inline comments

**Comment by vishnusomank** [comment]
> please find inline comments

@Ankurk99 sorry about that I had an old config file that's why it was removed. I've updated it according to your comment

**Review by seswarrajan — APPROVED** [review]
LGTM

**Review by Ankurk99 — APPROVED** [review]
LGTM!
